components/Events/EventInfo.js

```diff
@@ -0,0 +1,31 @@
+import React, { useState, useEffect } from "react";
+import EventCard from "./EventCard";
+import styles from "./Event.module.css";
+
+const DisplayEventInfo = (props) => {
+  const [eventData, setEventData] = useState([]);
+
+  const getEventDataHandler = async () => {
+    try {
+      setEventData(props.result);
+    } catch (error) {
+      alert("fetch failed!!!");
+      console.log(error);
+      window.location = "/500";
+    }
+  };
+  // call get data handler in the useEffect hook which runs once when the component is mounted.
+  useEffect(() => {
```

Comment:
I will get `Unhandled Runtime Error Error: Too many re-renders. React limits the number of renders to prevent an infinite loop.`, when I remove useEffect.🧐

Reply:
try to remove `useEffect` and `getEventDataHandler` and just connect prop data to `EventCard` component?

Reply:
If you call `getEventDataHandler` without `useEffect`, it will call `setState` and it will lead to the infinite re-render. Because it's a promise with a component that it re-render when there is `setState` called, then but component calls `setState` again and so on. So that's why we used `useEffect` which is for to decide when to execute `setState`. But since we've moved the fetching logic to `setServersideProp`, we don't even need anything here.